FEXCore/Source/Interface/Core/JIT/MemoryOps.cpp

```diff
@@ -1536,6 +1536,17 @@ DEF_OP(StoreMem) {
       str(Src.D(), MemSrc);
       break;
     }
+    case IR::OpSize::f80Bit: {
+      LOGMAN_THROW_A_FMT(HostSupportsSVE128, "80-bit stores are only supported with SVE128 support");
+      const auto MemSrc = GenerateSVEMemOperand(OpSize, MemReg, Op->Offset, Op->OffsetType, Op->OffsetScale);
+      // We need to generate the predicate register to copy the 10-byte value
+      // FIXME: don't hardcode p5
+      ARMEmitter::PRegister preg = ARMEmitter::PReg::p5;
+      mov(TMP1, 10);
+      whilelt(ARMEmitter::SubRegSize::i8Bit, preg, ARMEmitter::XReg::zr, TMP1);
```

Comment:
Can't generating the predicate use something like `ptrue(ARMEmitter::SubRegSize::i16Bit, preg, ARMEmitter::PredicatePattern::SVE_VL5` and then use `st1h<ARMEmitter::SubRegSize::i8Bit>(Src.Z(), preg, MemSrc);` ?
If it works then the predicate generation is 1 instruction (which is zero-cycle latency on A720), and then a single store.

Reply:
Maybe I am misunderstanding but VL5 is `0b00101`. The predicate needs to be setup with 0b1010 == 0xa, which wouldn't work.

Reply:
oh wait, but there is a pattern that would work, `  SVE_VL32 = 0b01010,`. OK - let me give it a go.

Reply:
Wrote a quick test, looks like my idea would work?
```cpp
#include <cstdio>
#include <cstdint>
#include <string.h>

__attribute__((naked))
void DoSVEStore(const void *in, void *out) {
	__asm volatile(R"(
ptrue p0.h, vl5;
ldr q0, [x0];
st1h z0.h, p0, [x1];
ret;
	)"
	::: "v0", "memory", "cc");
}

static const uint64_t Data[2] = {
	0x4142434445464748,
	0x5152535455565758,
};

static uint8_t DataOutSVE[16]{};
static uint8_t DataOutMemcpy[16]{};
void Do80BitStore(const void *in, void *out) {
	memcpy(out, in, 10);
}

int main() {
	DoSVEStore(Data, DataOutSVE);
	Do80BitStore(Data, DataOutMemcpy);

	fprintf(stderr, "      SVE | CPY  | Matches?\n"); 
	for (size_t i = 0; i < 16; ++i) {
	fprintf(stderr, "%02zd:  0x%02x | 0x%02x | %s\n", i, DataOutSVE[i], DataOutMemcpy[i], DataOutSVE[i] == DataOutMemcpy[i] ? "Yes" : "No");
	}
	return 0;
}
```

Output:
```bash
      SVE | CPY  | Matches?
00:  0x48 | 0x48 | Yes
01:  0x47 | 0x47 | Yes
02:  0x46 | 0x46 | Yes
03:  0x45 | 0x45 | Yes
04:  0x44 | 0x44 | Yes
05:  0x43 | 0x43 | Yes
06:  0x42 | 0x42 | Yes
07:  0x41 | 0x41 | Yes
08:  0x58 | 0x58 | Yes
09:  0x57 | 0x57 | Yes
10:  0x00 | 0x00 | Yes
11:  0x00 | 0x00 | Yes
12:  0x00 | 0x00 | Yes
13:  0x00 | 0x00 | Yes
14:  0x00 | 0x00 | Yes
15:  0x00 | 0x00 | Yes
```

Also tested putting the edge of the store ten bytes before the end of a page and confirmed it is only storing 10-bytes

Reply:
Thanks - I see in ptrue you're using p0.h rather than p0.b and then vl5. Maybe I haven't truly understood how these patterns work but your example helps.